Find a functionally equivalent code:android.widget.SuggestionsAdapter.getDrawableFromResourceValue:COMMENT
Method Modifier: private     
Comment:/**
 * Gets a drawable given a value provided by a suggestion provider.
 *
 * This value could be just the string value of a resource id
 * (e.g., "2130837524"), in which case we will try to retrieve a drawable from
 * the provider's resources. If the value is not an integer, it is
 * treated as a Uri and opened with
 * {@link ContentResolver#openOutputStream(android.net.Uri, String)}.
 *
 * All resources and URIs are read using the suggestion provider's context.
 *
 * If the string is not formatted as expected, or no drawable can be found for
 * the provided value, this method returns null.
 *
 * @param drawableId a string like "2130837524",
 * "android.resource://com.android.alarmclock/2130837524",
 * or "content://contacts/photos/253".
 * @return a Drawable, or null if none found
 */

@@ -13,19 +13,19 @@
             return drawable;
         }
         // Not cached, find it by resource ID
-        drawable = mProviderContext.getResources().getDrawable(resourceId);
+        drawable = mProviderContext.getDrawable(resourceId);
         // Stick it in the cache, using the URI as key
         storeInIconCache(drawableUri, drawable);
         return drawable;

